src/selectors/clusterSelectors.js

```diff
@@ -13,8 +13,20 @@ import {
 import { createDeepEqualSelector, typeWithoutSuffix } from './selectorUtils';
 
 // Regular selectors
-const selectClusterById = (state, props) => {
-  return state.entities.clusters.items[props.cluster.id];
+export const selectClusterById = (state, id) => {
+  return state.entities.clusters.items[id];
+};
+
+const selectOrganizationClusterNames = (state) => {
+  const clusters = state.entities.clusters.items;
+  const clusterIds = Object.keys(clusters);
+
+  return clusterIds
+    .filter((id) => clusters[id].owner === state.main.selectedOrganization)
+    .reduce(
```

Comment:
We can do the filtering in `reduce`'s lambda body, so we won't do the extra loop

Reply:
I've been about to not use `reduce`, just for readibility. If we add the check inside I think this will be less readable...

Reply:
To me this is equally as readable, and we're not executing any extra loop

```javascript
const selectOrganizationClusterNames = (state) => {
  const { selectedOrganization } = state.main;
  const clusters = state.entities.clusters.items;
  const clusterIDs = Object.keys(clusters);

  let cluster = {};

  return clusterIDs.reduce(
    (accumulator, id) => {
      cluster = clusters[id];

      if (cluster.owner !== selectedOrganization) {
        return accumulator;
      }

      return [...accumulator, { id, name: cluster.name }];
    },
    [],
  );
};
```

Reply:
I thinks that this is less readable, less semantic and less elegant. And from a performance point of view, we are not looping through a huge array here, so I really think it makes no difference 🤷‍♂